webpack/components/extensions/HostDetails/Tabs/HostTracesActions.js

```diff
@@ -1,12 +1,47 @@
-import { API_OPERATIONS, get } from 'foremanReact/redux/API';
-import { HOST_TRACES_KEY } from './HostTracesConstants';
+import { API_OPERATIONS, get, put } from 'foremanReact/redux/API';
+import { propsToCamelCase } from 'foremanReact/common/helpers';
+import { addToast } from 'foremanReact/redux/actions/toasts';
+import { HOST_TRACES_KEY, RESOLVE_HOST_TRACES_TASK_KEY } from './HostTracesConstants';
 import { foremanApi } from '../../../../services/api';
+import { getResponseErrorMsgs, pluralize } from '../../../../utils/helpers';
 
-const getHostTraces = (hostId, params) => get({
+const errorToast = (error) => {
+  const message = getResponseErrorMsgs(error.response);
+  return message;
+};
+
+export const getHostTraces = (hostId, params) => get({
   type: API_OPERATIONS.GET,
   key: HOST_TRACES_KEY,
   url: foremanApi.getApiUrl(`/hosts/${hostId}/traces`),
   params,
 });
 
-export default getHostTraces;
+export const resolveHostTraces = (hostId, params, dispatch) => {
+  const { traceIds } = propsToCamelCase(params);
+  return put({
+    type: API_OPERATIONS.PUT,
+    key: RESOLVE_HOST_TRACES_TASK_KEY,
+    url: foremanApi.getApiUrl(`/hosts/${hostId}/traces/resolve`),
+    handleSuccess: (response) => {
+      const { data: { id } } = response;
+      const traceCount = Number(traceIds.length);
+      dispatch({
+        type: 'TOASTS_ADD',
+        payload: {
+          key: id,
+          message: {
+            type: 'success',
+            message: `Restarting ${pluralize(traceCount, 'trace')}.`,
+            link: {
+              children: 'View task',
+              href: `/foreman_tasks/tasks/${id}`,
```

Comment:
This took me like 2 hours to figure out.. I didn't know about `webpack/scenes/Tasks/helpers.js` :sob:  I will update this

Reply:
K, yeah I'd like to see the dispatch call made into it's own function.

Reply:
Turns out this same thing can also be accomplished in a single line
```js
handleSuccess: response => renderTaskStartedToast(response.data),
```
(or close enough to the same thing, at least.) So I'm doing that instead.